analysis/estimate/notebooks/compare-estimate-consensus.Rmd

```diff
@@ -0,0 +1,340 @@
+
+Let's first look at what the consensus cell types that we have for this data are.
+For human readability, we'll look at the annotations, not ontologies, for this.
+
+```{r}
+all_celltypes <- celltype_data |>
+  purrr::map(
+    \(cell_df) dplyr::pull(cell_df, consensus_annotation) |> unique()
+  ) |>
+  purrr::reduce(union)
+
+all_celltypes
+```
+
+* We can use the `immune_df` data frame to identify immune cell types to use in comparisons
+* Since ESTIMATE defines stromal as non-cancer and non-immune (aka "everything else"), we'll define all non-immune and non-neural cell types as stromal cells
+  * An important caveat to bear in mind that is that two of our projects, SCPCP000006 (Wilms tumor) and SCPCP000017 (osteosarcoma), will have cancer cell types with a high degree of similarity to several stromal cell types, which can make teasing these categories apart challenging.
+  This is particularly relevant for mesangial cells in SCPCP000006; we are classifying these cell types as "stromal" but they may be highly similar to the cancer cells in those samples
+
+```{r}
+stromal_celltypes <- c(
+  "chondrocyte", 
+  "adipocyte", 
+  "endothelial cell",
+  "epithelial cell",              
+  "smooth muscle cell", 
+  "muscle cell", 
+  "blood vessel endothelial cell", 
+  "stromal cell", 
+  "pericyte",                     
+  "mesangial cell", 
+  "fibroblast"
+)
+
+immune_celltypes <- immune_df$consensus_annotation
+```
+  
+The following consensus cells are _not_ currently considered in either the immune or bulk group.
+
+```{r}
+other_celltypes <- all_celltypes[!(all_celltypes %in% c(stromal_celltypes, immune_celltypes))]
+other_celltypes
+```
+
+Next, we'll calculate proportions of these cell types across samples.
+
+```{r}
+celltype_category_df <- celltype_data |>
+  purrr::map(
+    \(cell_df) {
+        cell_df |>
+        dplyr::mutate(
+          estimate_category = dplyr::case_when(
+            consensus_annotation %in% immune_celltypes ~ "immune",
+            consensus_annotation %in% stromal_celltypes ~ "stromal", 
+            .default = "other"
+          )) |>
+        dplyr::count(estimate_category) |>
+        dplyr::mutate(frac = n/sum(n)) |>
+        dplyr::select(-n)
+    }
+  ) |>
+  purrr::list_rbind(names_to = "sample_id") |>
+  # we need to add in project ids
+  dplyr::inner_join(dplyr::select(id_df, project_id, sample_id)) |>
+  dplyr::mutate(estimate_category = forcats::fct_relevel(estimate_category, "other", "stromal", "immune"))
+```
+
+What do we have?
+
+
+```{r fig.height=12, fig.width=8}
+ggplot(celltype_category_df) + 
+  aes(x = sample_id, y = frac, fill = estimate_category) +
+  geom_col() + 
+  # bright color to help make stromal more visible
+  scale_fill_manual(values = c("gray85", "navy", "dodgerblue")) + 
+  facet_wrap(vars(project_id), scales = "free", ncol = 1) + 
+  scale_y_continuous(expand = c(0,0)) +
+  labs(y = "fraction of cells in sample") +
+  theme(axis.text.x = element_blank()) 
+```
+
+
```

Comment:
Let's add some comments about the immune cells. To be honest, I don't really trust the stromal scores or stromal cell type distributions as much as I do the immune cells, so I would focus on that.